Build error due `-Werror=class-memaccess` in GCC 8.1

The warning: `-Werror=class-memaccess`

How to reproduce:

- Have a GCC 8 (possibly 8.1)
- Set up a cmake build manually (make a subdirectory, `cmake ..` in it)
- `make`

Terminal output:

```
.../wabt> cd mybuild/
.../wabt/mybuild> cmake ..
-- The C compiler identification is GNU 8.1.0
-- The CXX compiler identification is GNU 8.1.0
-- Check for working C compiler: /usr/lib/ccache/bin/cc
-- Check for working C compiler: /usr/lib/ccache/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/lib/ccache/bin/c++
-- Check for working CXX compiler: /usr/lib/ccache/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for alloca.h
-- Looking for alloca.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for snprintf
-- Looking for snprintf - found
-- Looking for sysconf
-- Looking for sysconf - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Check size of size_t
-- Check size of size_t - done
-- Using prebuilt re2c lexer
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.15", minimum required is "2.7") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/marvin/dev/rust/wasm/wabt/mybuild
marvin@tp-marvin-arch ~/d/r/w/w/mybuild> make
Scanning dependencies of target libgtest
[  1%] Building CXX object CMakeFiles/libgtest.dir/third_party/gtest/googletest/src/gtest-all.cc.o
[  2%] Linking CXX static library liblibgtest.a
[  2%] Built target libgtest
Scanning dependencies of target libwabt
[  3%] Building CXX object CMakeFiles/libwabt.dir/src/token.cc.o
[  5%] Building CXX object CMakeFiles/libwabt.dir/src/opcode.cc.o
[  6%] Building CXX object CMakeFiles/libwabt.dir/src/error-handler.cc.o
[  7%] Building CXX object CMakeFiles/libwabt.dir/src/hash-util.cc.o
[  9%] Building CXX object CMakeFiles/libwabt.dir/src/filenames.cc.o
[ 10%] Building CXX object CMakeFiles/libwabt.dir/src/string-view.cc.o
[ 11%] Building CXX object CMakeFiles/libwabt.dir/src/ir.cc.o
[ 12%] Building CXX object CMakeFiles/libwabt.dir/src/expr-visitor.cc.o
[ 14%] Building CXX object CMakeFiles/libwabt.dir/src/lexer-source.cc.o
[ 15%] Building CXX object CMakeFiles/libwabt.dir/src/lexer-source-line-finder.cc.o
[ 16%] Building CXX object CMakeFiles/libwabt.dir/src/wast-parser-lexer-shared.cc.o
[ 18%] Building CXX object CMakeFiles/libwabt.dir/src/prebuilt/wast-lexer-gen.cc.o
[ 19%] Building CXX object CMakeFiles/libwabt.dir/src/wast-parser.cc.o
[ 20%] Building CXX object CMakeFiles/libwabt.dir/src/type-checker.cc.o
[ 22%] Building CXX object CMakeFiles/libwabt.dir/src/validator.cc.o
[ 23%] Building CXX object CMakeFiles/libwabt.dir/src/binary-reader.cc.o
[ 24%] Building CXX object CMakeFiles/libwabt.dir/src/binary-reader-logging.cc.o
[ 25%] Building CXX object CMakeFiles/libwabt.dir/src/binary-writer.cc.o
[ 27%] Building CXX object CMakeFiles/libwabt.dir/src/binary-writer-spec.cc.o
[ 28%] Building CXX object CMakeFiles/libwabt.dir/src/binary-reader-ir.cc.o
[ 29%] Building CXX object CMakeFiles/libwabt.dir/src/binding-hash.cc.o
[ 31%] Building CXX object CMakeFiles/libwabt.dir/src/wat-writer.cc.o
[ 32%] Building CXX object CMakeFiles/libwabt.dir/src/interp.cc.o
[ 33%] Building CXX object CMakeFiles/libwabt.dir/src/binary-reader-interp.cc.o
[ 35%] Building CXX object CMakeFiles/libwabt.dir/src/apply-names.cc.o
[ 36%] Building CXX object CMakeFiles/libwabt.dir/src/generate-names.cc.o
[ 37%] Building CXX object CMakeFiles/libwabt.dir/src/resolve-names.cc.o
[ 38%] Building CXX object CMakeFiles/libwabt.dir/src/binary.cc.o
[ 40%] Building CXX object CMakeFiles/libwabt.dir/src/color.cc.o
[ 41%] Building CXX object CMakeFiles/libwabt.dir/src/common.cc.o
[ 42%] Building CXX object CMakeFiles/libwabt.dir/src/config.cc.o
[ 44%] Building CXX object CMakeFiles/libwabt.dir/src/feature.cc.o
[ 45%] Building CXX object CMakeFiles/libwabt.dir/src/leb128.cc.o
[ 46%] Building CXX object CMakeFiles/libwabt.dir/src/literal.cc.o
[ 48%] Building CXX object CMakeFiles/libwabt.dir/src/option-parser.cc.o
[ 49%] Building CXX object CMakeFiles/libwabt.dir/src/stream.cc.o
[ 50%] Building CXX object CMakeFiles/libwabt.dir/src/tracing.cc.o
[ 51%] Building CXX object CMakeFiles/libwabt.dir/src/utf8.cc.o
[ 53%] Linking CXX static library libwabt.a
[ 53%] Built target libwabt
Scanning dependencies of target wabt-unittests
[ 54%] Building CXX object CMakeFiles/wabt-unittests.dir/src/test-circular-array.cc.o
[ 55%] Building CXX object CMakeFiles/wabt-unittests.dir/src/test-intrusive-list.cc.o
[ 57%] Building CXX object CMakeFiles/wabt-unittests.dir/src/test-literal.cc.o
[ 58%] Building CXX object CMakeFiles/wabt-unittests.dir/src/test-string-view.cc.o
[ 59%] Building CXX object CMakeFiles/wabt-unittests.dir/src/test-filenames.cc.o
[ 61%] Building CXX object CMakeFiles/wabt-unittests.dir/src/test-utf8.cc.o
[ 62%] Building CXX object CMakeFiles/wabt-unittests.dir/src/test-wast-parser.cc.o
[ 63%] Building CXX object CMakeFiles/wabt-unittests.dir/third_party/gtest/googletest/src/gtest_main.cc.o
[ 64%] Linking CXX executable wabt-unittests
[ 64%] Built target wabt-unittests
Scanning dependencies of target wabt-unittests-copy-to-bin
[ 64%] Built target wabt-unittests-copy-to-bin
Scanning dependencies of target hexfloat_test
[ 66%] Building CXX object CMakeFiles/hexfloat_test.dir/src/literal.cc.o
[ 67%] Building CXX object CMakeFiles/hexfloat_test.dir/src/test-hexfloat.cc.o
[ 68%] Building CXX object CMakeFiles/hexfloat_test.dir/third_party/gtest/googletest/src/gtest_main.cc.o
[ 70%] Linking CXX executable hexfloat_test
[ 70%] Built target hexfloat_test
Scanning dependencies of target wat-desugar
[ 71%] Building CXX object CMakeFiles/wat-desugar.dir/src/tools/wat-desugar.cc.o
[ 72%] Linking CXX executable wat-desugar
[ 72%] Built target wat-desugar
Scanning dependencies of target wat-desugar-copy-to-bin
[ 72%] Built target wat-desugar-copy-to-bin
Scanning dependencies of target spectest-interp
[ 74%] Building CXX object CMakeFiles/spectest-interp.dir/src/tools/spectest-interp.cc.o
/home/marvin/dev/rust/wasm/wabt/src/tools/spectest-interp.cc: In function ‘wabt::interp::Result spectest::DefaultHostCallback(const wabt::interp::HostFunc*, const wabt::interp::FuncSignature*, wabt::Index, wabt::interp::TypedValue*, wabt::Index, wabt::interp::TypedValue*, void*)’:
/home/marvin/dev/rust/wasm/wabt/src/tools/spectest-interp.cc:828:58: error: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘struct wabt::interp::TypedValue’; use assignment or value-initialization instead [-Werror=class-memaccess]
   memset(out_results, 0, sizeof(TypedValue) * num_results);
                                                          ^
In file included from /home/marvin/dev/rust/wasm/wabt/src/tools/spectest-interp.cc:31:
/home/marvin/dev/rust/wasm/wabt/src/interp.h:148:8: note: ‘struct wabt::interp::TypedValue’ declared here
 struct TypedValue {
        ^~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/spectest-interp.dir/build.make:63: CMakeFiles/spectest-interp.dir/src/tools/spectest-interp.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:333: CMakeFiles/spectest-interp.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

--- a/src/tools/spectest-interp.cc
+++ b/src/tools/spectest-interp.cc
@@ -825,7 +825,8 @@ static interp::Result DefaultHostCallback(const HostFunc* func,
                                           Index num_results,
                                           TypedValue* out_results,
                                           void* user_data) {
-  memset(out_results, 0, sizeof(TypedValue) * num_results);
+  memset(static_cast<void*>(out_results), 0,
+                            sizeof(TypedValue) * num_results);
   for (Index i = 0; i < num_results; ++i)
     out_results[i].type = sig->result_types[i];
 

--- a/src/tools/wasm-interp.cc
+++ b/src/tools/wasm-interp.cc
@@ -201,7 +201,8 @@ class WasmInterpHostImportDelegate : public HostImportDelegate {
                                       Index num_results,
                                       TypedValue* out_results,
                                       void* user_data) {
-    memset(out_results, 0, sizeof(TypedValue) * num_results);
+    memset(static_cast<void*>(out_results), 0,
+                              sizeof(TypedValue) * num_results);
     for (Index i = 0; i < num_results; ++i)
       out_results[i].type = sig->result_types[i];
 

